returns min possible size vol

boyamurthy opened the pull request:
allowing to create volume with minimum supported size. 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/1305 |

# Checklist:

- [X] Have you run format,vet & lint checks against your submission?
- [X] Have you made sure that the code compiles?
- [X] Did you run the unit & integration tests successfully?
- [ ] Have you maintained at least 90% code coverage?
- [ ] Have you commented your code, particularly in hard-to-understand areas
- [ ] Have you done corresponding changes to the documentation
- [X] Did you run tests in a real Kubernetes cluster?
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

Unit test with new changes. 

![image](https://github.com/dell/csi-powermax/assets/92081029/faec2420-11ec-43cd-ac9d-1861e7da44fb)

delldubey reviewed (approved):
lgtm

shanmydell commented on service/controller.go:
```diff
@@ -1501,9 +1501,9 @@ func (s *service) validateVolSize(ctx context.Context, cr *csi.CapacityRange, sy
-		return 0, status.Errorf(
-			codes.OutOfRange,
-			"bad capacity: requested minimum size (%d bytes) is less than the minimum volume size (%d bytes)", minSizeBytes, MinVolumeSizeBytes)
+		log.Warningf("bad capacity: requested minimum size (%d bytes) is less than the minimum volume size (%d bytes)", minSizeBytes, MinVolumeSizeBytes)
```
Requested size (%d bytes) is less than the minimum volume size (%d bytes) supported by PowerMax..

boyamurthy replied:
updated.

shanmydell commented on service/controller.go:
```diff
@@ -1501,9 +1501,9 @@ func (s *service) validateVolSize(ctx context.Context, cr *csi.CapacityRange, sy
-			codes.OutOfRange,
-			"bad capacity: requested minimum size (%d bytes) is less than the minimum volume size (%d bytes)", minSizeBytes, MinVolumeSizeBytes)
+		log.Warningf("bad capacity: requested minimum size (%d bytes) is less than the minimum volume size (%d bytes)", minSizeBytes, MinVolumeSizeBytes)
+		log.Warning("provisioning vol with minimum volume size...")
```
Proceeding with minimum volume size supported by PowerMax Array ...

boyamurthy replied:
updated.

gallacher reviewed (changes requested):
Please document the testing that was performed to validate this fix.  Have unit tests been put in place to test various scenarios around volume size?  If not, please ensure they are in. Not just happy path.  Thanks.

boyamurthy commented:
> Please document the testing that was performed to validate this fix. Have unit tests been put in place to test various scenarios around volume size? If not, please ensure they are in. Not just happy path. Thanks.

Hi @gallacher , added the unit tests to cover the various volume sizes and added the screen shot for the same. Thank you.

nidtara reviewed (approved):
lgtm
